db/database.go

```diff
@@ -186,21 +196,26 @@ func (db *Database) GetAllSwaps() ([]*swap.Info, error) {
 
 	var swaps []*swap.Info
 	for iter.Valid() {
-		key := iter.Key()
+		id := iter.Key()
 
 		// if the key becomes longer than 32, we're not iterating over swaps
-		if len(key) > idLength {
+		if len(id) > idLength {
 			break
 		}
 
 		// value is the encoded swap
 		encodedSwap := iter.Value()
 		s, err := swap.UnmarshalInfo(encodedSwap)
 		if err != nil {
-			return nil, err
+			log.Warnf("removing invalid swap info with offerID=0x%X from database: %s", id, err)
+			log.Warnf("invalid offer JSON was: %s", string(encodedSwap))
+			if err = db.swapTable.Del(id[:]); err != nil {
+				return nil, fmt.Errorf("failed to remove invalid offer from database: %w", err)
+			}
```

Comment:
I'm not sure what the behavior should be here. Deserialization failed, so we don't know if the invalid swap was ongoing or not. Options are:
* Remove the entry as coded above
* Don't return the entry but leave it in the database (giving warnings on every startup)
* Prevent startup

Reply:
What is the data consistency "agreement" between the `swapTable` and the `offerTable`?  When we remove an offer from the `offerTable`, does any matching Swap Info entry with the same ID need to removed as well?

Reply:
hm, I guess this would be a bad case we would like to prevent with versioning. potentially, all swap keys would need to be prefixed by `swap/<version>` so that deserialization can't fail (unless the db got corrupted or something)  I would remove it, we should try to ensure this case doesn't happen.

once the offer is taken it's deleted from the db and a `swap.Info` struct is created and written to the db. however this does raise a problem (unrelated to this PR), if a swap refunds/aborts, the `swap.Info` stays in the db (with the exit status saved), but the offer is also re-added to the db.

Reply:
Can you add an issue for the problem mentioned above? I'd create it, but I'm not sure I can describe it adequately.

Reply:
yes will do!

Reply:
#298